pkg/helper/client.go

```diff
@@ -0,0 +1,84 @@
+package helper
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1"
+)
+
+const nmstateCommand = "nmstatectl"
+const namespace = "default"
+
+func nmstatectl(arguments ...string) (string, error) {
+	cmd := exec.Command(nmstateCommand, arguments...)
+	var outb, errb bytes.Buffer
+	cmd.Stdout = &outb
+	cmd.Stderr = &errb
+	if err := cmd.Run(); err != nil {
+		return "", fmt.Errorf("Failed to execute nmstatectl show: '%v'", err)
+	}
+	return outb.String(), nil
+}
+
+func GetNodeNetworkState(client client.Client, nodeName string, nodeNetworkState *nmstatev1.NodeNetworkState) error {
+	nodeNetworkStateKey := types.NamespacedName{
+		Namespace: namespace,
+		Name:      nodeName,
+	}
+
+	return client.Get(context.TODO(), nodeNetworkStateKey, nodeNetworkState)
+}
+
+func CreateNodeNeworkState(client client.Client, nodeName string) error {
+	nodeNetworkState := nmstatev1.NodeNetworkState{}
```

Comment:
why don't you populate meta and spec right here?

Reply:
Done.